### PR TITLE
Feat/account endpoint

### PR DESCRIPTION
**Pull request by manihagh**
- [X] Account balance lookup
- [x] Historical balance lookup

Implements: #3

### CODE REVIEW

File: src/account/account.service.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      
+      if (!block || (block.number !== blockNumber)) {        
+        return Errors.BLOCK_NOT_FOUND;  
+      }
+      blockIdentifier.index = block.number
+      blockIdentifier.hash = block.hash
+
+      const balance = await provider.getBalance(accountAddress, blockIdentifier.index)      
+      balances.push({
+        value: utils.formatEther(balance),
```
Comment (kosecki123) [inline src/account/account.service.ts:28]: I think we should return `balance` as lowest denomination value.

File: src/account/account.service.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    } else {    
+      // latest balance loookup
+      const blockNumber = await provider.getBlockNumber()      
+      
+      const block = await provider.getBlock(blockNumber)
+      blockIdentifier.index = block.number
+      blockIdentifier.hash = block.hash
+
+      const balance = await provider.getBalance(accountAddress)
```
Comment (kosecki123) [inline src/account/account.service.ts:40]: This actually will not return consistent results, these calls are not atomic, so between executing `getBlockNumber()` and `getBalance()` the block change can happen.

I suggest to simply this whole function as follows:

```
const block = await provider.getBlock(blockNumber ?? "latest");
const balance = await provider.getBalance(accountAddress, block.number);

return {
"block_identifier": new BlockIdentifier(block.number, block.hash),
"balances": [{value: balance, currency: Currency.EWT}]
}
```

File: src/account/account.service.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    let blockIdentifier = new PartialBlockIdentifier()
+    let balances = []
+    
+    if (blockNumber){      
+      // historical balance lookup            
+      const block = await provider.getBlock(blockNumber)      
+      
+      if (!block || (block.number !== blockNumber)) {        
+        return Errors.BLOCK_NOT_FOUND;  
+      }
```
Comment (kosecki123) [inline src/account/account.service.ts:22]: Maybe better to move this to PartialBlockIdentifier `validate` function?

**Review by kosecki123 — CHANGES_REQUESTED** [review]

File: src/account/account.service.ts (hunk 3 above)
Comment (manihagh) [reply src/account/account.service.ts:22]: Agree, moved

File: src/account/account.service.ts (hunk 2 above)
Comment (manihagh) [reply src/account/account.service.ts:40]: Yes, I missed that when changed from iterative async

**Review by kosecki123 — APPROVED** [review]

**Pull request merged**